### PR TITLE
⚡  REFACTOR: [bulk/main] relocate logs into report creator

### PR DESCRIPTION
**Pull request by chancesmith**
## overall purpose
Filtering contacts is causing a side effect for reporting sake. This could be unclear to the developer.

## changes made
- move all reporting into createStatusReport
- pass contacts into createStatusReport
- modify contacts with `skip: boolean` for generating and reporting the right contacts

## screenshot
Screen Shot 2022-12-29 at 11 08 43 AM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ interface Contact {
   "Office Phone*": string;
   "Mobile Phone": string;
   "Calendly Link": string;
+  skip?: boolean;
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:15]: add skip prop to help generate the right contacts and help with reporting

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -74,6 +75,8 @@ async function generateSignatures(
   contacts: Contact[]
 ) {
   contacts.forEach(async (contact) => {
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:77]: skip contact if didn't have required data

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -160,10 +163,30 @@ async function zipUpFile() {
...
+  // 2. report in console
+  console.log(
+    `${skippedRows.length ? "🙅‍♂️" : "👍"} Signatures:`,
+    processedRows.length,
+    `processed and`,
+    skippedRows.length,
+    `skipped`
+  );
+
+  // 3. report in file
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:189]: so we can move report consts away from being a global, here we move all reporting into this report function

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -173,15 +196,16 @@ async function createStatusReport({
...
-  return new Promise((resolve: (arg: Contact[]) => void, reject) => {
-    return fs.promises.readFile(path).then((fileData) => {
-      parse(fileData, { columns: true }, function (err, rows) {
-        resolve(rows as Contact[]);
-      });
+  return new Promise(async (resolve: (arg: Contact[]) => void) => {
+    const file = await fs.promises.readFile(path);
+
+    parse(file, { columns: true }, function (err, rows) {
+      resolve(rows as Contact[]);
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:208]: remove .then()

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -253,21 +260,16 @@ async function main() {
...
-    `${reportLog.skippedRows.length ? "🙅‍♂️" : "👍"} Signatures:`,
-    reportLog.processedRows.length,
-    `processed and`,
-    reportLog.skippedRows.length,
-    `skipped`
-  );
+    .map((contact) => ({
+      ...contact,
+      skip: !checkRequiredFields(contact),
+    }));
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:268]: change filter to map so we can track the contacts for generating and reporting

**Review by adamnitti — APPROVED** [review]

**Pull request merged**